frontend/src/api/actions/applications.js

```diff
@@ -88,11 +91,36 @@ export const deleteApplication = validatedApiDispatcher({
 // search for and return the isolated state associated with `reducer`. This is not
 // a standard redux function.
 export const localStoreSelector = applicationsReducer._localStoreSelector;
-export const applicationsSelector = createSelector(
+export const _applicationsSelector = createSelector(
     localStoreSelector,
     state => state._modelData
 );
 
+export const applicationsSelector = createSelector(
+    [_applicationsSelector, applicantsSelector, positionsSelector],
+    (applications, applicants, positions) => {
+        if (applications.length === 0) {
+            return [];
+        }
+
+        const applicantsById = arrayToHash(applicants);
+        const positionsById = arrayToHash(positions);
+
+        return applications.map(
+            ({ application_preferences, applicant_id, ...rest }) => ({
```

Comment:
These are `position_preferences`. In general, I don't like replacing a field with a recomputed one in a selector. I think it's better that they have different names. For example, replacing `applicant_id` with `applicant` is fine because the fields have different names. This case kinda walks the line, since you're modifying what's inside the preferences...

Reply:
Oh, when you said "it's an application's preferences for positions", I thought you meant it is ok to keep it as `application_preferences`. Not sure if you meant we should only name it as `position_preferences` in here but keep it as `application_preferences` in the API and mock data.

Reply:
If it's in the application, it should be `position_preferences`

Reply:
Also, not sure how we feel about this, but I think it might be a good idea if we recompute each object key from snake case to camel case (probably write a helper function and call that function in `createSelector`):
1. Better coding style as it will mostly have only camel cases in each component
2. As we are centralizing the model data processing, when each individual component receive its' `props`, it should not concern/know whether or not the data is coming from API.
3. It will also help with the situation here. The selector will then have `applicationPreferences ` instead of `application_preferences` so it is easier to separate them.
Of course, that also mean we need to recompute it to use snake case before we send it to API, but I think it is a good practice in general unless there is some other reason why we chose do it the way it is right now

Reply:
I don't like this idea, but we can talk about it Friday.